tests/test_modeling_common.py

```diff
@@ -398,7 +398,9 @@ class CopyClass(base_class):
                 model_slow_init = base_class_copy.from_pretrained(tmpdirname, _fast_init=False)
 
                 for key in model_fast_init.state_dict().keys():
-                    max_diff = (model_slow_init.state_dict()[key] - model_fast_init.state_dict()[key]).sum().item()
+                    if random_key_to_del.endswith(key):
+                        continue
```

Comment:
I think this is reasonable, but need @sgugger to confirm.

Reply:
No, the test is there to check that the corresponding weight is initialized the same way with slow and fast init. This would make the test less useful.

Reply:
Alright, I couldn't see the reason for which the weights (for that removed key) should have the same values with 2 different initialization methods. As these are random ops, and not running in the same sequence in these 2 cases.

And if we don't expect the same values but just close values, it's not very clear to me how the threshold is determined.

(But I didn't really go through the whole logic around this part ...)

Reply:
I might be wrong about what the test is doing but I thought it was checking that. Though it would require setting the seed before both load to work. @patrickvonplaten could you shed some light here since you wrote it?

Reply:
Setting seed is a necessary but not sufficient condition in this case. As we change the way of initialization, which will change the position (in the sequence of random operations performed by `torch`) where a particular weight being initialized (here for the deleted key) - I believe.

If we want to obtain equal weight values with both init. methods, we will have to set the same seed just before the target weight(s). However, I feel this is not really trackable and not worth the effort.

Reply:
The naming might not have been spot-on here. In short the mechanism and the rationale behind the test here is to test that the initialization not **only** works for initializing whole parts of the model (e.g. the model head, or the whole cross attention layer), but **also** when the checkpoint is missing a single weight / parameter. It's an edge cast test.

 E.g. what we want to test here is the following scenario: The bias of the first feed-forward layer of the 5th block of BERT is removed from the state-dict, then this single bias parameter will be initialized when loading the state dict with `from_pretrained(...)` -> now this initialization should be the same for both the fast and slow init method. IMO we should definitely keep this line because removing it makes the test worse. 

Does that make sense? 
This should be unrelated to the seed because if I remember correctly the `_init_weights` weights is overwritten by a deterministic `self._mock_init_weights` function, so there is no randomness in play here.